src/routes/ExercisesScreens.tsx

```diff
@@ -24,17 +25,24 @@ interface ExercisesScreenProps {
 }
 
 const ExercisesScreen = ({ route, navigation }: ExercisesScreenProps): JSX.Element => {
-  const { discipline } = route.params
-  const { title } = discipline
+  const { params } = route
+  const { disciplineTitle } = params
 
-  const Header = <Title title={title} description={childrenDescription(discipline)} />
+  const load = useCallback(async (params: ExercisesParams) => {
+    if (params.documents) {
+      return params.documents
+    }
+    return loadDocuments(params.discipline)
+  }, [])
+  const { data: documents, error, loading, refresh } = useLoadAsync(load, params)
+
+  const Header = documents && <Title title={disciplineTitle} description={wordsDescription(documents.length)} />
 
   const handleNavigation = (item: Exercise): void => {
-    if (item.title === labels.exercises.wordChoice.title && discipline.numberOfChildren < MIN_WORDS) {
-      Alert.alert(labels.exercises.wordChoice.errorWrongModuleSize)
-    } else {
+    if (documents) {
```

Comment:
Oke I see you changed the behaviour. I'm fine with this, also may help when adding the unlocking stuff.

Reply:
I removed this because I thought we need it to be able to do the exercise for favorites if there are less then three favorites. Since this is not required anymore I could revert it, but I think this solution makes more sense/is easier. But I am open to reverting if I should

Reply:
Keep it as it is, it's fine or even better.